### PR TITLE
deps: use go1.17 and remove GO111MODULE

### PR DESCRIPTION
**Pull request by chenrui333**
- deps: update to use go1.17
- remove `GO111MODULE` setting as go1.16 defaults to go mod build

### CODE REVIEW

**Comment by chenrui333** [comment]
signed the CLA.

**Review by dani-santos-code — APPROVED** [review]
hi, @chenrui333 ! Thank you for your contribution. It looks good!

**Comment by dani-santos-code** [comment]
can we merge this @genevieveluyt ?

File: go.mod [hunk 1]
```diff
@@ -12,4 +14,47 @@ require (
 	k8s.io/client-go v0.22.4
 )
 
-go 1.16
+require (
```
Comment (genevieveluyt) [inline go.mod:17]: Are there supposed to be two `require` blocks? Can you please run `go mod tidy`?

**Review by genevieveluyt — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (genevieveluyt) [reply go.mod:17]: Interesting, thanks for context!

**Comment by genevieveluyt** [comment]
@chenrui333 there are a couple other places that still reference 1.16 if you wouldn't mind updating those as well

- [ ] https://github.com/Shopify/kubeaudit/blob/main/goreleaser.Dockerfile#L1
- [ ] https://github.com/Shopify/kubeaudit/blob/main/Dockerfile#L1
- [ ] https://github.com/Shopify/kubeaudit/blob/main/Makefile#L13
- [ ] https://github.com/Shopify/kubeaudit/blob/main/README.md?plain=1#L50

**Pull request closed without merging**